### PR TITLE
Add dependency on blockchain-k-plugin so KEVM can run with Mantis

### LINKED ISSUE #146: Integrate KEVM with Mantis
We intend to set up the EVM semantics so that it uses the blockchain-k-plugin we wrote for IELE in order to build a VM server that can actually mine blocks.

### PR DESCRIPTION
**Pull request by dwightguth**
This PR should make all the Mantis ethereum-tests tests pass.

Fixes #146

### CODE REVIEW

**Comment by ehildenb** [comment]
@dwightguth initial review:

-   Does the submodule need to be at `/plugin`? Is it ok to hide it in `.build/plugin`? I would prefer it.
-   Please put `&&` on the next line in the Makefile when doing split lines, indented from the previous line (as the style was before the change). This makes it easy to delete a step by deleting a single line.

I will need more time to read the K code in detail.

**Comment by dwightguth** [comment]
@ehildenb I fixed the issue with the lines in the Makefile, but I think that the submodule should be left where it is, because unlike the submodules under .build, the plugin isn't really a dependency, it's actually part of the code of the project, but it lives in a submodule simply because that code is shared with another project.

File: evm-node.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+    syntax Int ::= #getBlockhash(Int) [function, hook(MANTIS.getBlockhash)]
+ // -----------------------------------------------------------------------
+    rule <k> BLOCKHASH N => #getBlockhash(N) ~> #push ... </k> <mode> NORMAL </mode>
+      requires N >=Int 0 andBool N <Int 256
+    rule <k> BLOCKHASH N => 0 ~> #push ... </k> <mode> NORMAL </mode>
+      requires N <Int 0 orBool N >=Int 256
+
+    syntax EthereumSimulation ::= runVM(iscreate: Bool, to: Int, from: Int, code: String, args: String, value: Int, gasprice: Int, gas: Int, beneficiary: Int, difficulty: Int, number: Int, gaslimit: Int, timestamp: Int, unused: String)
+
```
Comment (ehildenb) [inline evm-node.md]: I've never seen this syntax for K. Is it some sort of record syntax?

Just it just enable the line below (matching on a subset of arguments?)

File: evm-node.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
+```{.k .node}
```
Comment (ehildenb) [inline evm-node.md]: This file needs to be documented properly (or at least at all).

File: evm-node.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    rule <k> #endCreate => W ... </k> <wordStack> W : WS </wordStack>
+
+    syntax KItem ::= vmResult(return: String,gas: Int,refund: Int,status: Int,selfdestruct: List,logs: List,AccountsCell, touched: List)
+    syntax KItem ::= extractConfig(GeneratedTopCell) [function]
+ // -----------------------------------------------------------
+    rule extractConfig(<generatedTop>... <output> OUT </output> <gas> GAVAIL </gas> <refund> REFUND </refund> <k> STATUS:Int </k> <selfDestruct> SD </selfDestruct> <log> LOGS </log> <accounts> ACCTS </accounts> <touchedAccounts> TOUCHED </touchedAccounts> ... </generatedTop>) => vmResult(#unparseByteStack(OUT),GAVAIL,REFUND,STATUS,Set2List(SD),LOGS,<accounts> ACCTS </accounts>, Set2List(TOUCHED))
+
+    syntax String ::= contractBytes(WordStack) [function]
+ // -----------------------------------------------------
+    rule contractBytes(WS) => #unparseByteStack(WS)
```
Comment (ehildenb) [inline evm-node.md]: `contractBytes == #unparseByteStack`? Is the indirection for some compatibility layer?

File: evm-node.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    imports EVM
+    imports K-REFLECTION
+    imports COLLECTIONS
+
+    syntax AccountCode ::= "#unloaded"
+
+    syntax Int ::= #getBalance(Int) [function, hook(MANTIS.getBalance)]
+                 | #getNonce(Int) [function, hook(MANTIS.getNonce)]
+    syntax Bool ::= #isCodeEmpty(Int) [function, hook(MANTIS.isCodeEmpty)]
+                  | #accountExists(Int) [function, hook(MANTIS.accountExists)]
```
Comment (ehildenb) [inline evm-node.md]: I'm not a huge fan of the hooks being `MANTIS.*`, because they shouldn't be Mantis specific, should they?

File: Makefile [hunk 5]
```diff
@@ -116,24 +125,52 @@ defn: $(defn_files)
...
+DLLEXT=cma
+OCAMLC=c
+LIBFLAG=-a
+else
+EXT=cmx
+LIBEXT=cmxa
+DLLEXT=cmxs
+OCAMLC=opt -O3
+LIBFLAG=-shared
+endif
```
Comment (ehildenb) [inline Makefile]: Can this be indented (perhaps with spaces if not tabs)? Looks strange.

File: evm-node.md (hunk 3 above)
Comment (dwightguth) [reply evm-node.md]: yes

File: evm-node.md (hunk 1 above)
Comment (dwightguth) [reply evm-node.md]: yes

File: Makefile (hunk 5 above)
Comment (dwightguth) [reply Makefile]: fixed

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
@ehildenb please re-review.

**Pull request merged**